tutorials/3d/fps_tutorial/part_one.rst

```diff
@@ -71,213 +86,243 @@ First, go open the project settings and go to the "Input Map" tab. You'll find s
-The reason behind this is so we rotate ``Player`` on the ``Y`` axis, and ``Rotation_helper`` on
+Another thing to notice is how many nodes are children of ``Rotation_Helper``. This is because
+``Rotation_Helper`` contains all of the nodes we want to rotate on the ``X`` axis (up and down).
+The reason behind this is so we can rotate ``Player`` on the ``Y`` axis, and ``Rotation_helper`` on
 the ``X`` axis.
 
 .. note:: If we did not use ``Rotation_helper`` then we'd likely have cases where we are rotating
           both the ``X`` and ``Y`` axes at the same time. This can lead to undesirable results, as we then
           could rotate on all three axes in some cases.
+          
+          See :ref:`using transforms <doc_using_transforms>` for more information
 
 _________
 
 Attach a new script to the ``Player`` node and call it ``Player.gd``.
 
-Lets program our player by adding the ability to move around, look around with the mouse, and jump.
+Let's program our player by adding the ability to move around, look around with the mouse, and jump.
 Add the following code to ``Player.gd``:
 
 ::
 
     extends KinematicBody
 
-    const norm_grav = -24.8
+    const GRAVITY = -24.8
     var vel = Vector3()
     const MAX_SPEED = 20
     const JUMP_SPEED = 18
-    const ACCEL = 3.5
+    const ACCEL= 4.5
 
+    var dir = Vector3()
+    
     const DEACCEL= 16
     const MAX_SLOPE_ANGLE = 40
-
+    
     var camera
-    var camera_holder
-
-    # You may need to adjust depending on the sensitivity of your mouse
-    const MOUSE_SENSITIVITY = 0.05
-
-    var flashlight
-
+    var rotation_helper
+    
+    var MOUSE_SENSITIVITY = 0.05
+    
     func _ready():
-        camera = $Rotation_helper/Camera
-        camera_holder = $Rotation_helper
+        camera = $Rotation_Helper/Camera
+        rotation_helper = $Rotation_Helper
         
         Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
-        
-        flashlight = $Rotation_helper/Flashlight
-
+    
     func _physics_process(delta):
-        var dir = Vector3()
+        process_input(delta)
+        process_movement(delta)
+    
+    func process_input(delta):
+        
+        # ----------------------------------
+        # Walking
+        dir = Vector3()
         var cam_xform = camera.get_global_transform()
-
+        
+        var input_movement_vector = Vector2()
+        
         if Input.is_action_pressed("movement_forward"):
-            dir += -cam_xform.basis.z.normalized()
+            input_movement_vector.y += 1
         if Input.is_action_pressed("movement_backward"):
-            dir += cam_xform.basis.z.normalized()
+            input_movement_vector.y -= 1
         if Input.is_action_pressed("movement_left"):
-            dir += -cam_xform.basis.x.normalized()
+            input_movement_vector.x -= 1
         if Input.is_action_pressed("movement_right"):
-            dir += cam_xform.basis.x.normalized()
-
+            input_movement_vector.x = 1
```

Comment:
Shouldn't this be `+= 1` ?

Reply:
Yeah, I think so.

Reply:
Ah, okay. Then yeah, that’s fine with me. Thanks!